### PR TITLE
[5.2] Added sqlserver odbc connection support

### PR DESCRIPTION
**Pull request by eddy8**
#13264

### CODE REVIEW

File: src/Illuminate/Database/Connectors/SqlServerConnector.php [hunk 1]
```diff
@@ -45,12 +45,29 @@ protected function getDsn(array $config)
...
 
     /**
+     * Get the DSN string for a ODBC connection.
+     *
+     * @param  array  $config
+     * @return string
+     */
+    protected function getOdbcDsn(array $config)
+    {
+        if (isset($config['ODBC_DATA_SOURCE_NAME'])) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: that naming convention is not the same as the rest of the config

Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: it also doesn't actually work properly

**Comment by taylorotwell** [comment]
What if odbc is an availble driver but they want to use the sqlsrv string?

**Comment by eddy8** [comment]
we can check driver config param in `config/database.php` file.
`'sqlsrv' => [
            'driver'   => 'sqlsrv',
            'host'     => env('DB_HOST', 'localhost'),
            'database' => env('DB_DATABASE', 'forge'),
            'username' => env('DB_USERNAME', 'forge'),
            'password' => env('DB_PASSWORD', ''),
            'charset'  => 'utf8',
            'prefix'   => '',
        ],`

**Pull request closed without merging**